app/src/marketplace/components/MarkdownEditor/index.jsx

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import React, { useState, useCallback } from 'react'
+import cx from 'classnames'
+
+import styles from './markdownEditor.pcss'
+
+type Props = {
+    placeholder?: string,
+    onChange?: (string) => void,
+    className?: string,
+}
+
+const MarkdownEditor = ({ placeholder, onChange, className }: Props) => {
+    const [text, setText] = useState('')
+    const [wordCount, setWordCount] = useState(0)
+
+    const onTextChange = useCallback((e: SyntheticInputEvent<EventTarget>) => {
+        const { value } = e.target
+        setText(value)
+        const trimmedValue = value.trim()
+        setWordCount(trimmedValue.length > 0 ? trimmedValue.split(/\s+/g).length : 0)
+
+        if (onChange != null) {
+            onChange(value)
+        }
+    }, [onChange])
```

Comment:
Would it make sense to separate the word count into an effect, something like:
```
    const onTextChange = useCallback((e: SyntheticInputEvent<EventTarget>) => {
        const { value } = e.target
        setText(value)

        if (onChange != null) {
            onChange(value)
        }
    }, [onChange])

    useEffect(() => {
        const trimmedValue = text.trim()
        setWordCount(trimmedValue.length > 0 ? trimmedValue.split(/\s+/g).length : 0)
    }, [text])
```

Reply:
True. It would be cleaner this way 👍